--- a/openLuup/server.lua
+++ b/openLuup/server.lua
@@ -75,6 +75,7 @@
 local wsapi     = require "openLuup.wsapi"              -- WSAPI connector for CGI processing
 local tables    = require "openLuup.servertables"       -- mimetypes and status_codes
 local vfs       = require "openLuup.virtualfilesystem"
+local uci       = require "uci"
 
 --  local log
 local function _log (msg, name) logs.send (msg, name or ABOUT.NAME) end
@@ -122,7 +123,7 @@
 end
 
 -- http://forums.coronalabs.com/topic/21105-found-undocumented-way-to-get-your-devices-ip-address-from-lua-socket/
-local myIP = (
+local myIP = uci:cursor().get("network","lan","ipaddr") or (
   function ()    
     local mySocket = socket.udp ()
     mySocket:setpeername ("42.42.42.42", "424242")  -- arbitrary IP and PORT
@@ -655,7 +656,7 @@
 -- returns list of utility function(s)
 -- 
 local function start (port, backlog)
-  local server, msg = socket.bind ('*', port, backlog or BACKLOG) 
+  local server, msg = socket.bind (myIP, port, backlog or BACKLOG) 
    
   -- new client connection
   local function server_incoming (server)
